packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -479,16 +484,28 @@ export async function startTransformationJob(uploadId: string) {
         getLogger().info(`CodeTransformation: jobId: ${jobId}`)
     } catch (error) {
         getLogger().error(`CodeTransformation: ${CodeWhispererConstants.failedToStartJobNotification}`, error)
-        if ((error as Error).message.includes('too many active running jobs')) {
+        const errorMessage = (error as Error).message
+        if (errorMessage.includes('too many active running jobs')) {
             transformByQState.setJobFailureErrorNotification(
                 CodeWhispererConstants.failedToStartJobTooManyJobsNotification
             )
             transformByQState.setJobFailureErrorChatMessage(
                 CodeWhispererConstants.failedToStartJobTooManyJobsChatMessage
             )
+        } else if (errorMessage.includes('Lines of Code limit breached')) {
```

Comment:
Would it be better to create custom Error types and switch on the type rather than looking at strings? If strings get updated in one place but not everywhere, then these lines of code break

Reply:
We'd still have to check the Strings though right? This String comes directly from backend.

Reply:
Not a blocker: I think @JuliaABurch is asking about semantic readability and maybe a small separation of concerns. The code might read better for developers if it was something like

`else if (error === LinesOfCodeLimitBreachedError)` and you abstract the "logic" of the string checking in the error types we create.